main.tf

```diff
@@ -136,6 +97,13 @@ resource "aws_iam_role_policy_attachment" "cloudwatchagentserverpolicy" {
   role       = join("", aws_iam_role.main.*.name)
 }
 
+## Managed Policy to allow ssm agent to communicate with SSM Manager
+resource "aws_iam_role_policy_attachment" "ssm" {
+  count      = var.create_instance_iam_role ? 1 : 0
```

Comment:
This is not being restricted by the `var.install_ssm_agent = true/false` variable which means it will always attach these permissions so if the instance has the agent even if the `var.install_ssm_agent = false` is set.

Also please add to the backlog because this is the only way to guarantee we can't use the ssm agent:
`if var. install_ssm_agent = false uninstall the ssm agent from the instance Linux/Windows.`

Reply:
Added